smithy-docgen/build.gradle.kts

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+plugins {
+    id("smithy.module-conventions")
+    id("smithy.integ-test-conventions")
+}
+
+description = "This module contains support for generating API documentation based on Smithy models."
+
+extra["displayName"] = "Smithy :: DocGen"
+extra["moduleName"] = "software.amazon.smithy.docgen"
+
+dependencies {
+    implementation(project(":smithy-build"))
+    implementation(project(":smithy-codegen-core"))
+    implementation(project(":smithy-linters"))
+    implementation(project(":smithy-model"))
```

Comment:
Model and utils should be unnecessary here b/c they are already api deps of smithy-build
https://github.com/smithy-lang/smithy/blob/main/smithy-build/build.gradle.kts#L16C1-L17C34

Similarly, codegen-core defines `model` and `build` as api deps, so this can be simplified to just `smithy-codegen-core` and `smithy-linters`